physics/discrete_trajectory.hpp

```diff
@@ -182,20 +182,56 @@ class DiscreteTrajectory : public Forkable<DiscreteTrajectory<Frame>,
   std::int64_t timeline_size() const override;
 
  private:
-  struct Downsampling {
+  class Downsampling {
+   public:
+    Downsampling(std::int64_t max_dense_intervals,
+                 Length tolerance,
+                 TimelineConstIterator start_of_dense_timeline,
```

Comment:
Remove this parameter and set the member to `begin()` always.

Reply:
No, this is used for deserialization too.